noria-server/dataflow/src/state/persistent_state.rs

```diff
@@ -511,9 +439,9 @@ impl PersistentState {
     }
 
     // Filters out secondary indices to return an iterator for the actual key-value pairs.
-    fn all_rows(&self) -> impl Iterator<Item = (Box<[u8]>, Box<[u8]>)> {
+    fn all_rows(&self) -> impl Iterator<Item = (Box<[u8]>, Box<[u8]>)> + '_ {
```

Comment:
The compiler suggested that I added `'_` here but I have no idea if that's a sane thing to do—here's the error:

```rust
error: lifetime may not live long enough
   --> noria-server/dataflow/src/state/persistent_state.rs:442:27
    |
442 |     fn all_rows(&self) -> impl Iterator<Item = (Box<[u8]>, Box<[u8]>)> {
    |                 -         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ opaque type requires that `'1` must outlive `'static`
    |                 |
    |                 let's call the lifetime of this reference `'1`
help: to allow this impl Trait to capture borrowed data with lifetime `'1`, add `'_` as a constraint
    |
442 |     fn all_rows(&self) -> impl Iterator<Item = (Box<[u8]>, Box<[u8]>)> + '_ {
```

Reply:
Oh, yeah, that's probably fine --- `impl Trait` by default assumes the lifetime of all input arguments, so `'_` is a way of indicating that it shouldn't capture the lifetime of `self`.